### PR TITLE
[Feature] Suggest recipes based on user's inventory

### PR DESCRIPTION
**Pull request by Lohenrex**
## Description
MoguMogu's killer feature. Being a user able to upload their current ingredient inventory, this module finds recipes that can be made with those ingredients and shows them to the user in a handy view.

## Screenshots
The oracle in action.
The oracle in action

### CODE REVIEW

**Review by itsmeurbi — APPROVED** [review]
Seems to be legit! 🐓
Well done!

File: app/views/home/_oracle.html.erb [hunk 1]
```diff
@@ -0,0 +1,54 @@
+<%= tag.h4 "We couldn't find a recipe that matches your inventory. :(", class: "subtitle has-text-centered" unless @doable_recipes.length > 0 %>
```
Comment (luisAzcuaga) [inline app/views/home/_oracle.html.erb:1]: You can also use `empty?` e.g: `if @doable_recipes.empty?`. Same effect, tho.

File: app/models/recipe.rb [hunk 2]
```diff
@@ -10,4 +10,6 @@ class Recipe < ApplicationRecord
   has_one_attached :picture do |attachable|
     attachable.variant :thumb, resize_to_limit: [100, 100]
   end
+
+  scope :get_recipe_which_has_ingredient, ->(ingredient) { where("recipes.ingredients->'needed' @> ?", [{ name: ingredient }].to_json) }
```
Comment (luisAzcuaga) [inline app/models/recipe.rb:14]: It's kinda redundant having 'recipe' in the scope name.
I'd rather go with `scope :with_ingredient`. So, when you consume the scope would look like:
```ruby
Recipe.with_ingredient(item)
```

**Review by luisAzcuaga — APPROVED** [review]
LGTM!
I know this was pretty hard, hope we could have achieve the jsonb query tho :c. 
Anyway, code is lit 🔥 
Left some minor suggestions, but you have my approval!

File: app/models/recipe.rb (hunk 2 above)
Comment (itsmeurbi) [reply app/models/recipe.rb:14]: Agree on that 👆🏻
We're already talking about recipes so...

File: app/views/home/_oracle.html.erb (hunk 1 above)
Comment (itsmeurbi) [reply app/views/home/_oracle.html.erb:1]: Even more clear 👆🏻

**Pull request merged**